models/checks.go

```diff
@@ -0,0 +1,849 @@
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/flanksource/kommons"
+	v1 "k8s.io/api/core/v1"
+)
+
+type ICheck interface {
```

Comment:
@yashmehrotra had to rename this interface from `Check`. It was in `canary-checker/pkg/api/external`